src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -666,6 +661,7 @@ private LocalSymbol GetDeclaredLocal(CSharpSyntaxNode declarationSyntax, SyntaxT
                         LocalSymbol adjustedLocal;
                         if (Compilation.NullableSemanticAnalysisEnabled)
                         {
+                            _analyzedVariableTypesOpt ??= new ConcurrentDictionary<LocalSymbol, LocalSymbol>();
```

Comment:
Probably should use `Interlocked.CompareExchange()` to avoid populating multiple dictionaries.

Reply:
I'm going to not lazily initialize this. @agocke mentioned that our lock-free code policy at this point is to get the entire team to review. If the concurrent dictionary shows up in a trace, we can lazily initialize at that point.

---
In reply to: [306527725](https://github.com/dotnet/roslyn/pull/37322#discussion_r306527725) [](ancestors = 306527725)

Reply:
If you feel strongly about this, I can use a `Lazy<T>` on the concurrent dictionary, but that feels like overkill imo.

---
In reply to: [306933090](https://github.com/dotnet/roslyn/pull/37322#discussion_r306933090) [](ancestors = 306933090,306527725)

Reply:
Clarification: I didn't actually look at this code. The policy is we have to review new lock-free data structures together -- all existing patterns are fine. In this case, this would be equivalent to a lazy initialization, which is a well-established pattern.

The point of the policy wasn't to prevent us from using our standard (well-reviewed) lazy initialization patterns, just to prevent us from adding clever new patterns that have very subtle threading requirements.